tycho-integration-test/src/ot.rs

```diff
@@ -0,0 +1,62 @@
+use miette::{miette, IntoDiagnostic, Result};
+use opentelemetry::global;
+use opentelemetry_otlp::WithExportConfig;
+use opentelemetry_sdk::{propagation::TraceContextPropagator, runtime, trace, Resource};
+use serde::Deserialize;
+use tracing::{error, Subscriber};
+use tracing_subscriber::{
+    layer::SubscriberExt, registry::LookupSpan, util::SubscriberInitExt, EnvFilter, Layer,
+};
+
+#[derive(Debug, Clone, Deserialize)]
+pub struct TracingConfig {
+    pub otlp_exporter_endpoint: String,
+}
+
+/// Initialize tracing: apply an `EnvFilter` using the `RUST_LOG` environment variable to define the
+/// log levels, add a formatter layer logging trace events as JSON and on OpenTelemetry layer
+/// exporting trace data.
+pub fn init_tracing(config: TracingConfig) -> Result<()> {
+    global::set_text_map_propagator(TraceContextPropagator::new());
+
+    global::set_error_handler(|error| error!(error = format!("{error:#}"), "otel error"))
+        .map_err(|e| miette!("Failed to set error handler: {e}"))?;
+
+    let format = tracing_subscriber::fmt::format()
+        .with_level(true)
+        .with_target(false)
```

Comment:
Any context on why we do this in our crates? It's impossible to see where a log line is coming from with this setting.